system/database/DB_query_builder.php

```
@@ -899,6 +899,14 @@ protected function _wh_in($qb_key, $key = NULL, $values = NULL, $not = FALSE, $t
 		{
 			$wh_in = array_values($values);
 		}
+		
+		if (is_array($key))
+		{
+			$key = '('.implode(', ', $key).')';
+			$wh_in = array_map(function ($value) {
```

This makes the *assumption* that we have a multidimensional array, masking potential errors.  
It may've not been an array at all originally (previous `if` condition).

On another note, you're changing the docblock for this method, but not for the actual public methods ...